internal/controller/utils/nim.go

```diff
@@ -119,18 +119,27 @@ func GetAvailableNimRuntimes(logger logr.Logger) ([]NimRuntime, error) {
 }
 
 // ValidateApiKey is used for validating the given API key by retrieving the token and pulling the given custom runtime
-func ValidateApiKey(logger logr.Logger, apiKey string, runtime NimRuntime) error {
-	tokenResp, tokenErr := getRuntimeRegistryToken(logger, apiKey, runtime.Resource)
-	if tokenErr != nil {
-		return tokenErr
-	}
+func ValidateApiKey(logger logr.Logger, apiKey string, runtimes []NimRuntime) error {
+	for _, runtime := range runtimes {
+		tokenResp, tokenErr := getRuntimeRegistryToken(logger, apiKey, runtime.Resource)
+		if tokenErr != nil {
+			return tokenErr
+		}
 
-	manifestErr := attemptToPullManifest(logger, runtime, tokenResp)
-	if manifestErr != nil {
-		return manifestErr
-	}
+		manifestResp, manifestErr := attemptToPullManifest(logger, runtime, tokenResp)
+		if manifestErr != nil {
+			return manifestErr
+		}
 
-	return nil
+		switch manifestResp.StatusCode {
+		case http.StatusUnavailableForLegalReasons:
+			continue
+		case http.StatusOK:
+			return nil
+		}
+		break
```

Comment:
just replace `break` with line142? 
maybe even print what code is return now for better visibility

Reply:
> just replace break with line142?
Can you clarify, please? Line 142 is outside the context of the loop.

> maybe even print what code is return now for better visibility
Good catch. We were logging it before, and I accidentally removed that code in this PR. Resolved in https://github.com/opendatahub-io/odh-model-controller/pull/390/commits/350b7059f52457fb25a5bda236b3690bef38fb98.

Reply:
i was thinking if you wanna break the for...loop then hit the return error
you could directly return error here. 
or i misunderstood this intention?

Reply:
in general, i am fine with this change.
i let others make the finall call 
/LGTM

Reply:
> i was thinking if you wanna break the for...loop then hit the return error
you could directly return error here.

Oh yeah, I see what you mean now. This should work, but Go blocks the build for that, so I'll have to throw a `return nil` there. So I think what we have now is better, more clear.